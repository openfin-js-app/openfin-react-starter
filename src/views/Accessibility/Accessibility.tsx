import * as React from 'react';
import * as shortid from 'shortid';
import { connect } from 'react-redux';

import Button from '@material-ui/core/Button';
import Snackbar from '@material-ui/core/Snackbar';
import Typography from '@material-ui/core/Typography';

import { withStyles } from '@material-ui/core/styles';

import { Applcation, Window } from '@albertli/redux-openfin';

import { applicationNewSnackbar, applicationLaunchNewWindow } from '../../redux';

import { buttonStyle } from '../../assets/jss/openfin-starter';

const style = {
    primary: buttonStyle.primary,
    info: buttonStyle.info,
    success: buttonStyle.success,
    warning: buttonStyle.warning,
    danger: buttonStyle.danger,
    rose: buttonStyle.rose,
    white: buttonStyle.white,
    simple: buttonStyle.simple,
};

class AccessbilityView extends React.Component<any,any>{
    render(){

        const {
            classes,
            actions:{
                onOpenNewSelf,onOpenGoogle,
                handleOpenPrimarySnackBar, handleOpenSecondarySnackBar,
            }
        } = this.props;

        return(
            <React.Fragment>
                <Typography
                    variant={"title"} gutterBottom
                >
                    Accessibility view works
                </Typography>

                <Button size={"large"} variant={"contained"} color={"primary"} onClick={onOpenNewSelf}
                >Report</Button>
                <Button size={"large"} variant={"contained"} color={"secondary"} onClick={onOpenGoogle}
                >Google</Button>


                <Button size={"large"} variant={"contained"} color={"primary"} onClick={handleOpenPrimarySnackBar}
                >Snackbar 1</Button>
                <Button size={"large"} variant={"contained"} color={"secondary"} onClick={handleOpenSecondarySnackBar}
                >Snackbar 2</Button>

                <hr/>

                <Button size={"small"} variant={"contained"} className={classes.primary}
                >Primary</Button>
                <Button size={"small"} variant={"contained"} className={classes.info}
                >Info</Button>
                <Button size={"small"} variant={"contained"} className={classes.success}
                >Success</Button>
                <Button size={"small"} variant={"contained"} className={classes.warning}
                >Warning</Button>
                <Button size={"small"} variant={"contained"} className={classes.danger}
                >Danger</Button>
                <Button size={"small"} variant={"contained"} className={classes.rose}
                >Rose</Button>


            </React.Fragment>
        )
    }
}

export default connect(
    (state:any)=>({

    }),
    dispatch => ({
        actions:{
            onOpenNewSelf:()=>{
                dispatch(applicationLaunchNewWindow({
                    name:`openfin-react-starter-child-accessibility-${shortid.generate()}`,
                    url:'/childWindow/report',
                    frame:false,
                    resizable:true,
                    state:'normal',
                    autoShow:true,
                    callback:(window:any)=>{
                        console.log('onOpenNewSelf::callback', window);
                    }
                }));
            },
            onOpenGoogle:()=>{
                dispatch(applicationLaunchNewWindow({
                    name:`openfin-react-starter-child-google-${shortid.generate()}`,
                    url:'https://www.google.com/',
                    frame:true,
                    resizable:true,
                    state:'normal',
                    autoShow:true,
                    callback:(window:any)=>{
                        console.log('onOpenGoogle::callback', window);
                    }
                }));
            },
            handleOpenPrimarySnackBar:()=>{
                dispatch(applicationNewSnackbar({
                    message:'Message to primary snackbar',
                    variant:'primary',
                }))
            },
            handleOpenSecondarySnackBar:()=>{
                dispatch(applicationNewSnackbar({
                    message:'Message to warning snackbar',
                    variant:'warning',
                }))
            },
        }
    })
)(withStyles(style)(AccessbilityView));