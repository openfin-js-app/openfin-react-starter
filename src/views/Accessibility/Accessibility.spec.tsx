import * as React from 'react';
import { Provider } from 'react-redux';

import { createMuiTheme } from '@material-ui/core/styles';
import { ThemeProvider } from '@material-ui/styles';
import { createShallow, createMount } from '@material-ui/core/test-utils';
import Button from '@material-ui/core/Button';
import configurestore from 'redux-mock-store';

import Accessibility from './Accessibility';

const mockStore = configurestore();
const initialState = {};

const muiTheme = createMuiTheme({});
const store = mockStore(initialState);

describe('AccessibilityView',()=>{

    let shallow;
    let mount;

    beforeEach(() => {
        mount = createMount();
        shallow = createShallow();
    });

    afterEach(()=>{
        mount.cleanUp();
    });

    it('render and fire 8 actions when 8 btn clicked',()=>{
       const wrapper = mount(
           <Provider store={store}>
               <ThemeProvider theme={muiTheme}>
                <Accessibility/>
               </ThemeProvider>
           </Provider>
       );
       expect(wrapper.find(Button)).toHaveLength(9);
       wrapper.find(Button).forEach((button)=>{
           const props = button.props();
           if ( typeof props.onClick === 'function'){
               props.onClick();
           }
       });
       expect(store.getActions()).toHaveLength(9);
    })
});