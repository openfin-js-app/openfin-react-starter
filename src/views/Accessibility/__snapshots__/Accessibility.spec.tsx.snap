// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccessibilityView render and fire actions when clicked 1`] = `
Array [
  Object {
    "payload": Object {
      "autoShow": true,
      "callback": [Function],
      "frame": false,
      "name": "openfin-react-starter-child-accessibility-U_TMCRJXX",
      "resizable": true,
      "state": "normal",
      "url": "/childWindow/report",
    },
    "type": "APPLICATION_LAUNCH_NEW_WINDOW",
  },
  Object {
    "payload": Object {
      "autoShow": true,
      "callback": [Function],
      "frame": true,
      "name": "openfin-react-starter-child-google-pg9QE8jfug",
      "resizable": true,
      "state": "normal",
      "url": "https://www.google.com/",
    },
    "type": "APPLICATION_LAUNCH_NEW_WINDOW",
  },
  Object {
    "payload": Object {
      "message": "Message to primary snackbar",
      "variant": "primary",
    },
    "type": "APPLICATION_NEW_SNACKBAR",
  },
  Object {
    "payload": Object {
      "message": "Message to warning snackbar",
      "variant": "warning",
    },
    "type": "APPLICATION_NEW_SNACKBAR",
  },
]
`;
